python/pyspark/sql/session.py

```diff
@@ -695,9 +704,10 @@ def prepare(obj):
             prepare = lambda obj: obj
```

Comment:
Hm, here the verification is done at `prepare` once. After that you do it one more time at L483. Seems like it does twice when the schema is specified.

Reply:
You could just pass extra boolean, and do the verification only in the case at this line (when the verification is not done).

Reply:
> Seems like it does twice when the schema is specified.

When the schema is specified, `_inferSchema` and `_create_verified_converter ` will not be invoked 

https://github.com/apache/spark/blob/c561ee686551690bee689f37ae5bbd75119994d6/python/pyspark/sql/session.py#L490-493